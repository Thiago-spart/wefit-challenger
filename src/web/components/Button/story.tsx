/* eslint-disable import/no-default-export */
/* eslint-disable import/exports-last */
import type { ComponentStory, Meta } from "@storybook/react";

import { Button } from "./";

const meta: Meta<typeof Button> = {
	title: "Components/Button",
	component: Button,
};

const Template: ComponentStory<typeof Button> = props => <Button {...props} />;

export const Primary = Template.bind({});
Primary.args = {
	type: "button",
	children: "Click me",
	isFullWidth: false,
	schema: "primary",
	fontSize: "XS",
};

export const Secondary = Template.bind({});
Secondary.args = {
	type: "button",
	children: "Click me",
	isFullWidth: false,
	schema: "secondary",
	fontSize: "XS",
};

export default meta;
