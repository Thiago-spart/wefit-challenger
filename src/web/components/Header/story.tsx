/* eslint-disable import/no-default-export */
/* eslint-disable import/exports-last */
import type { ComponentStory, Meta } from "@storybook/react";

import { Header } from "./";

const meta: Meta<typeof Header> = {
	title: "Layout/Header",
	component: Header,
};

const Template: ComponentStory<typeof Header> = props => <Header {...props} />;

export const Default = Template.bind({});
Default.args = {
	cartAmount: 1,
};

export default meta;
