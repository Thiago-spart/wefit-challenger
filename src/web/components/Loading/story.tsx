/* eslint-disable import/no-default-export */
/* eslint-disable import/exports-last */
import type { ComponentStory, Meta } from "@storybook/react";

import { Loading } from "./";

const meta: Meta<typeof Loading> = {
	title: "Components/Loading",
	component: Loading,
};

const Template: ComponentStory<typeof Loading> = () => <Loading />;

export const Primary = Template.bind({});
Primary.args = {};

export default meta;
