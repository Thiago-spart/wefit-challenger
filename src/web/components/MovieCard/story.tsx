/* eslint-disable import/no-default-export */
/* eslint-disable import/exports-last */
import type { ComponentStory, Meta } from "@storybook/react";

import { MovieCard } from "./";

const meta: Meta<typeof MovieCard> = {
	title: "Components/MovieCard",
	component: MovieCard,
};

const Template: ComponentStory<typeof MovieCard> = props => (
	<MovieCard {...props} />
);

export const Default = Template.bind({});
Default.args = {
	handleAddProduct: () => {},
	imgAlt: "Some alt description",
	title: "Pele Forever",
	price: 9000,
	productAmount: 2,
	imgLink: "https://www.imagemhost.com.br/images/2022/07/10/the-batman.png",
};

export default meta;
