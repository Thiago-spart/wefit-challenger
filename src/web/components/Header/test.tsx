import { composeStories } from "@storybook/testing-react";
import { render, screen } from "@testing-library/react";

import * as stories from "./story";

const { Default } = composeStories(stories);

describe("Header component", () => {
	it("should match snapshot", () => {
		const { container } = render(<Default cartAmount={1} />);

		expect(container).toMatchSnapshot();
	});

	it("link on logo should redirect to home page", () => {
		render(<Default cartAmount={5} />);

		const linkElement = screen.getByRole("link", { name: "WeMovies" });

		expect(linkElement).toHaveAttribute("href", "/");
	});

	it("link on cart should redirect to Cart page", () => {
		render(<Default cartAmount={5} />);

		const linkElement = document.querySelector("[href='/cart']");

		expect(linkElement).toHaveAttribute("href", "/cart");
	});

	it("should have the correct amount of items", () => {
		const { container, getByText } = render(<Default cartAmount={5} />);

		expect(getByText("5 itens", container)).toBeTruthy();
	});
});
