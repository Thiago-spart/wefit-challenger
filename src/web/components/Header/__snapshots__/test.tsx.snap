// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<div>
  <header
    class="sc-bcXHqe lgugwa"
  >
    <a
      aria-details="Logo"
      class="sc-gswNZR kNnreG"
      href="/"
    >
      WeMovies
    </a>
    <a
      href="/cart"
    >
      <div
        class="sc-dkrFOg eNiMdm"
      >
        <div>
          <h4>
            Meu Carrinho
          </h4>
          <span
            aria-label="Items on your cart"
          >
            1 item
          </span>
        </div>
        <img
          alt="market cart"
          data-nimg="1"
          decoding="async"
          height="32"
          loading="lazy"
          src="/cart.svg"
          style="color: transparent;"
          width="32"
        />
      </div>
    </a>
  </header>
</div>
`;

exports[`Header component should match snapshot 1`] = `
<div>
  <header
    class="sc-bcXHqe lgugwa"
  >
    <a
      aria-details="Logo"
      class="sc-gswNZR kNnreG"
      href="/"
    >
      WeMovies
    </a>
    <a
      href="/cart"
    >
      <div
        class="sc-dkrFOg eNiMdm"
      >
        <div>
          <h4>
            Meu Carrinho
          </h4>
          <span
            aria-label="Items on your cart"
          >
            1 item
          </span>
        </div>
        <img
          alt="market cart"
          data-nimg="1"
          decoding="async"
          height="32"
          loading="lazy"
          src="/cart.svg"
          style="color: transparent;"
          width="32"
        />
      </div>
    </a>
  </header>
</div>
`;
